src/types/widget.ts

```diff
@@ -119,6 +118,31 @@ export type UiState = {
   [indexId: string]: IndexUiState;
 };
 
+export type RenderState = {
+  [indexId: string]: IndexRenderState;
+};
+
+export type IndexRenderState = Partial<{
+  searchBox: WidgetRenderState<
+    {
+      query: string;
+      refine(query: string): void;
+      clear(): void;
+      isSearchStalled: boolean;
+    },
+    {
+      queryHook?(query: string, refine: (query: string) => void);
+    }
```

Comment:
Will we need to repeat the types here? Could they be imported from the widgets maybe?

Reply:
SearchBox is not migrated to TypeScript. We'll reuse `*RendererOptions` for other widgets here.